crates/harp/src/routines.rs

```diff
@@ -27,16 +41,24 @@ pub unsafe fn r_register_routines() {
     }
 
     // Make sure we have an "empty" routine at the end.
-    R_ROUTINES.push(R_CallMethodDef {
+    add(R_CallMethodDef {
         name: std::ptr::null(),
         fun: None,
         numArgs: 0,
     });
 
+    // Now unwrap the definitions from our thread-safe type
+    let unwrapped: Vec<R_CallMethodDef> = R_ROUTINES
+        .lock()
+        .unwrap()
+        .iter()
+        .map(|def| def.inner)
+        .collect();
```

Comment:
I'm fairly surprised you dont get a warning about "dropping temporary value" since no one is holding the lock in a local variable here. I seem to get that all the time when doing stuff like this.